src/common/net_common.h

```diff
@@ -69,23 +70,20 @@ class NetCommon {
     }
 
     static bool GetLocalIP(std::string* ipstr) {
-        char hostname[128];
-        int ret = gethostname(hostname, sizeof(hostname));
-
-        if (ret == -1) {
-            LOG(INFO) << "get hostname failed!";
-            return false;
+        struct ifaddrs *ifAddr = NULL;
+        struct ifaddrs *ifa = NULL;
+        getifaddrs(&ifAddr);
+        for (ifa = ifAddr; ifa != NULL; ifa = ifa->ifa_next) {
+            if (!ifa->ifa_addr) {
+                continue;
+            }
+            if (ifa->ifa_addr->sa_family == AF_INET) {
+                break;
+            }
         }
-
-        struct hostent* t;
-        t = gethostbyname(hostname);
-
-        if (t == nullptr) {
-            LOG(INFO) << "get host ip failed!";
-            return false;
-        }
-
-        char* ip = inet_ntoa(*(struct in_addr*)t->h_addr_list[0]);
+        char ip[INET_ADDRSTRLEN];
+        struct in_addr adr = ((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+        inet_ntop(AF_INET, &adr, ip, INET_ADDRSTRLEN);
```

Comment:
I got `127.0.0.1` using this new approach

<img width="367" alt="image" src="https://github.com/opencurve/curve/assets/10327590/3bb23eb5-1c74-482b-97f5-b5d400b7ff78">

Reply:
ok，It's a long time, I will test it to check what happened.